biogeochem/EDPhysiologyMod.F90

```diff
@@ -2133,41 +2155,120 @@ end subroutine SeedIn
 
   ! ============================================================================
 
-  subroutine SeedDecay( litt )
+  subroutine SeedDecay( litt , currentPatch, bc_in )
     !
     ! !DESCRIPTION:
-    !  Flux from seed pool into leaf litter pool
+    ! 1. Flux from seed pool into leaf litter pool
+    ! 2. If the TRS with seedling dynamics is on (regeneration_model = 3)
+    !    then we calculate seedling mortality here (i.e. flux from seedling pool
+    !    (into leaf litter pool)   
     !
     ! !ARGUMENTS
     type(litter_type) :: litt
+    type(ed_patch_type), intent(in) :: currentPatch ! ahb added this
+    type(bc_in_type), intent(in) :: bc_in ! ahb added this    
     !
     ! !LOCAL VARIABLES:
     integer  ::  pft
+    real(r8) ::  seedling_layer_par          ! cumulative sum of PAR at the seedling layer (MJ)
+                                             ! over prior window of days defined by 
+                                             ! fates_trs_seedling_mort_par_timescale
+    real(r8) ::  seedling_light_mort_rate    ! daily seedling mortality rate from light stress
+    real(r8) ::  seedling_h2o_mort_rate      ! daily seedling mortality rate from moisture stress
+    real(r8) ::  seedling_mdds               ! moisture deficit days accumulated in the seedling layer
+   
     !----------------------------------------------------------------------
 
+    
+    ! 1. Seed mortality (i.e. flux from seed bank to litter)
+    
     ! default value from Liscke and Loffler 2006 ; making this a PFT-specific parameter
     ! decays the seed pool according to exponential model
     ! seed_decay_rate is in yr-1
     ! seed_decay is kg/day
     ! Assume that decay rates are same for all chemical species
 
-    do pft = 1,numpft
+    !=====================================================================================
+    do pft = 1,numpft 
+    
+       ! If the TRS is switched off or the pft can't get big enough to be considered a tree 
+       ! then use FATES default regeneration.
+       if ( regeneration_model == default_regeneration .or. &
+            prt_params%allom_dbh_maxheight(pft) < min_max_dbh_for_trees ) then
+
+       ! Default seed decay (TRS is off)
        litt%seed_decay(pft) = litt%seed(pft) * &
-            EDPftvarcon_inst%seed_decay_rate(pft)*years_per_day
+             EDPftvarcon_inst%seed_decay_rate(pft)*years_per_day
 
-       litt%seed_germ_decay(pft) = litt%seed_germ(pft) * &
```

Comment:
@adamhb this could explain differences.  The seed_germ_decay rate is removed for the default model, but not re-inserted elsewhere, only for the TRS model. I'll look to see if this line is found somewhere else

Reply:
Yes! Good catch, thank you!